tests/Builder/Stage/UnionWithStageTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder\Stage;
+
+use MongoDB\Builder\Accumulator;
+use MongoDB\Builder\Expression;
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\Builder\PipelineTestCase;
+
+use function array_map;
+use function array_merge;
+use function MongoDB\object;
+use function range;
+
+/**
+ * Test $unionWith stage
+ */
+class UnionWithStageTest extends PipelineTestCase
+{
+    public function testReport1AllSalesByYearAndStoresAndItems(): void
+    {
+        $pipeline = new Pipeline(...array_merge(
+            [
+                Stage::set(
+                    _id: '2017',
+                ),
+            ],
+            array_map(
+                fn ($year) => Stage::unionWith(
+                    coll: 'sales_' . $year,
+                    pipeline: new Pipeline(
+                        Stage::set(
+                            _id: (string) $year,
+                        ),
+                    ),
+                ),
+                range(2018, 2020),
+            ),
```

Comment:
Experimentation to use `array_map`. But array unpacking can't be used in the middle of 2 other positional arguments. So I had to unpack the result of an `array_merge`.

Reply:
The `Pipeline` class could expand lists of provided as argument.

```php
new Pipeline(
    Stage::set(/* ... */),
    array_map(/* ... */),
    Stage::sort(/* ... */),
);
```

Reply:
For the time being, I'd format it manually the way it was done in the second test. I know that's a bit more effort to write, the `...array_merge()` logic with `array_map` is a bit fancy to understand.

Reply:
Ok, reverted.